docs/data/data-grid/rows/LazyLoadingGrid.js

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { DataGridPro } from '@mui/x-data-grid-pro';
+import {
+  useDemoData,
+  getRealGridData,
+  getCommodityColumns,
+} from '@mui/x-data-grid-generator';
+
+async function sleep(duration) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, duration);
+  });
+}
+
+const loadServerRows = async (newRowLength) => {
+  const newData = await getRealGridData(newRowLength, getCommodityColumns());
+  // Simulate network throttle
+  await sleep(Math.random() * 100 + 100);
+
+  return newData.rows;
+};
+
+export default function LazyLoadingGrid() {
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 10,
+    maxColumns: 6,
+  });
+
+  const handleFetchRows = async (params, event, details) => {
+    const newRowsBatch = await loadServerRows(
+      params.lastRowToRender - params.firstRowToRender,
+    );
+
+    details.api.replaceRows(
+      params.firstRowToRender,
+      params.lastRowToRender,
+      newRowsBatch,
+    );
+  };
+
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGridPro
+        {...data}
+        hideFooterPagination
+        rowCount={50}
+        sortingMode="server"
+        filterMode="server"
+        rowsLoadingMode="server"
```

Comment:
I would be more precise than `server` here

https://www.ag-grid.com/javascript-data-grid/viewport/

AG-Grid calls it the *Viewport Row Model* which is one of the *Server-Side Data* models along with *Infinite Row Model* or *Server Side Row Model*.

Reply:
it's called `server` because we use the same terminology for the filter and sorting feature modes. It uses the type `GridFeatureMode`

Reply:
I agree. we can revisit this while doing the breaking changes for v6.